tools/pipelines/build-docs.yml

```diff
@@ -2,14 +2,15 @@
 # Licensed under the MIT License.
 
 # build-docs pipeline
```

Comment:
Since this pipeline is now effectively _only_ about building and publishing API docs artifacts, we should update these docs to reflect that (and move any other docs to the other pipeline as appropriate).

We may also want to reconsider the pipeline/naming templates of these things. Since the other pipeline is the more general docs build one, it may want to be `build-docs` (so we can keep the existing pipeline definitions in ADO), and this may want to be `build-api-docs-artifacts` (or something), for which we'll need to define new pipelines in ADO.